src/components/Questionnaire/QuestionnaireForm.tsx

```diff
@@ -646,8 +646,13 @@ export function QuestionnaireForm({
       const nonStructuredResponses = form.responses.filter(
         (response) => !response.structured_type,
       );
-
-      if (nonStructuredResponses.length > 0) {
+      const validResponses = nonStructuredResponses.filter(
+        (response) =>
+          response.values.length > 0 &&
+          !response.structured_type &&
+          response.values?.[0]?.value !== "",
+      );
+      if (validResponses.length > 0) {
```

Comment:
In this case, why even call submitBatch when there are no valid responses?

(ofc, assuming there are no structured from responses).